src/PowerShellEditorServices.Hosting/EditorServicesLoader.cs

```diff
@@ -0,0 +1,338 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using SMA = System.Management.Automation;
+using System.Runtime.InteropServices;
+
+#if CoreCLR
+using System.Runtime.Loader;
+#else
+using Microsoft.Win32;
+#endif
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// Class to contain the loading behavior of Editor Services.
+    /// In particular, this class wraps the point where Editor Services is safely loaded
+    /// in a way that separates its dependencies from the calling context.
+    /// </summary>
+    public sealed class EditorServicesLoader : IDisposable
+    {
+        private const int Net461Version = 394254;
+
+        private static readonly string s_psesDependencyDirPath = Path.GetFullPath(
+            Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+                "..",
+                "Common"));
+
+        /// <summary>
+        /// Create a new Editor Services loader.
+        /// </summary>
+        /// <param name="logger">The host logger to use.</param>
+        /// <param name="hostConfig">The host configuration to start editor services with.</param>
+        /// <param name="sessionFileWriter">The session file writer to write the session file with.</param>
+        /// <returns></returns>
+        public static EditorServicesLoader Create(
+            HostLogger logger,
+            EditorServicesConfig hostConfig,
+            ISessionFileWriter sessionFileWriter,
+            IReadOnlyCollection<IDisposable> loggersToUnsubscribe = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)